src/Illuminate/Database/Query/Builder.php

```diff
@@ -1922,12 +1922,20 @@ public function implode($column, $glue = '')
     }
 
     /**
-     * Determine if any rows exist for the current query.
+     * Determine if any rows exist for the current query and/or by id.
      *
+     * @param  int  $id
      * @return bool
      */
-    public function exists()
+    public function exists($id = null)
     {
+        // If an ID is passed to the method, we will set the where clause to check the
+        // ID to let developers simply and quickly check if a single row exists in
+        // the database without manually specifying the "where" clauses on the query.
+        if (! is_null($id)) {
+            $this->where('id', '=', $id);
+        }
+
```

Comment:
Please don't use a hard coded column name. Use the getkey method.

Reply:
Not sure how to do that - I'm likely missing something.

exists is a method on Query/Builder, not Eloquent.  getKey is a method on Model and you can instantiate Query/Builder without a model.  I think we would need a getKey method that pulls the key from the database (or Model iff exists)? Or there is one and I didn't find it.

Eg I can't do this:

`$this->where($this->getModel()->getKey(), '=', $id);`
because $this may not have a model:
`$result = DB::table('users')->exists(1);`

Reply:
Yes. getKey is only on Eloquent Builder.

Reply:
Okay, nevermind.

Reply:
@tillkruss your change request is still open.  Anything else you would like changed?  I have added an exists method to Eloquent/Builder to use the key defined on the Model.